benchmarks/memory/scripts/test.js

```diff
@@ -0,0 +1,101 @@
+const yargs = require(`yargs`)
+const util = require('util')
+const { exit } = require('process')
+const exec = util.promisify(require('child_process').exec)
+
+const args = yargs
+  .option(`memory`, {
+    default: '2g',
+    describe: `The memory limit for the docker container`,
+  })
+  .option(`num-nodes`, {
+    default: '1000',
+    describe: `The number of nodes to generate in the build`,
+  })
+  .option(`node-size`, {
+    default: '1m',
+    describe: `The rough size of each node generated in the build`,
+  }).parse()
+
+const hostExec = (cmd) => {
+  console.log(`\x1b[2m$ ${cmd}\x1b[0m`)
+  return exec(cmd)
+}
+
+async function runTest() {
+
+  const { stdout: testDockerId } = await hostExec(`./scripts/docker-get-id`)
+  if (testDockerId.trim()) {
+    console.log(`Docker container ${testDockerId.trim()} is currently running, shutting it down.`)
+    await hostExec(`./scripts/docker-stop`);
+  }
+
+  console.log(`\nStarting container with ${args.memory} memory.`)
+  const { stderr: dockerStartError } = await hostExec(`DOCKER_MEMORY_LIMIT=${args.memory} ./scripts/docker-start`)
+  if (dockerStartError) {
+    console.log("Encountered an error:")
+    console.log(dockerStartError)
+    exit(1)
+  }
+
+  let { stdout: dockerId } = await hostExec(`./scripts/docker-get-id`)
+  dockerId = dockerId.trim()
+
+  const dockerExec = (cmd, env = {}) => {
+    const envVars = Object.keys(env).map(key => `-e ${key}=${env[key]}`)
+    return hostExec(`docker exec ${envVars.join(' ')} ${dockerId} ${cmd}`)
+  }
+
+  console.log(`\nStarting test using container ${dockerId}.`)
+
+  let code = 139
+
+  // there's something buggy with the node/exec/docker-exec integration
+  // we're getting seg faults, so this loop is just a patch for that
+  // so we don't have to fix it right now
```

Comment:
How often does that happen? We shouldn't be getting flakes like that :S

Reply:
Maybe 1/5 times? Fairly often... I tried to debug it for ~30 minutes and couldn't find it. It has nothing to do with our build, but the `docker exec`. Still very confused by it.

I've accounted for this in our timings, so it only pulls the timing for the latest build. It'll slow the test run down a tiny bit, but doesn't affect much else. Just annoying. 😦